src/share/classes/com/sun/javatest/regtest/exec/CompileAction.java

```diff
@@ -364,7 +364,7 @@ public Status run() throws TestRunException {
             }
         }
 
-        if (script.enablePreview() && !seenEnablePreview) {
+        if (insertPos != -1 && script.enablePreview() && !seenEnablePreview) {
```

Comment:
The fix would be clearer and more self-explanatory if you used the boolean `runJavac`. instead of the expression `insertPos != -1`.  It is already the case that `runJavac` implies `insertPos` is not `-1`.

Reply:
Argh, missed that variable. Agreed, that would be better. Fixed.